### PR TITLE
Adds support for redis sentinel deployments

### PR DESCRIPTION
**Pull request by RuriRyan**
New RedisSentinelJedisProvider class which uses the
JedisSentinelPool to do all the magic.
Also a new config option to choose the redis_sentinel as DB type.

### CODE REVIEW

**Comment by stevenctl** [comment]
Do you have an example of the configuration and redis deployment you would use with this?

File: redis-persistence/src/main/java/com/netflix/conductor/jedis/RedisSentinelJedisProvider.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        jedisPool = new JedisSentinelPool(configuration.getClusterName(), sentinels, poolConfig);
+    }
+
+    @Override
+    public JedisCommands get() {
+        return jedisPool.getResource();
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
```
Comment (cyzhao) [inline redis-persistence/src/main/java/com/netflix/conductor/jedis/RedisSentinelJedisProvider.java]: Don't rely on finalize to close the pool, as a matter of fact, finalize is deprecated in Java 9 and newer versions.

**Comment by cyzhao** [comment]
Please add unit tests.

**Comment by RuriRyan** [comment]
I will add unit tests later. But I still would like to hear some opinions about the implementation.

**Comment by apanicker-nflx** [comment]
Please add some comment or ReadMe about the configuration that is to be used with this deployment.

**Pull request merged**